error.php

```diff
@@ -291,13 +291,15 @@
     'oci-collection.max' => 'ocicollection.max',
     'oci-collection.size' => 'ocicollection.size',
     'oci-collection.trim' => 'ocicollection.trim',
-    'class.allow-dynamic-properties.php' => 'class.allowdynamicproperties.php',
-    'class.return-type-will-change.php' => 'class.returntypewillchange.php',
-    'class.sensitive-parameter.php' => 'class.sensitiveparameter.php',
+    'splstack.setiteratormode' => 'spldoublylinkedlist.setiteratormode',
+    'splqueue.setiteratormode' => 'spldoublylinkedlist.setiteratormode',
+    'class.allow-dynamic-properties' => 'class.allowdynamicproperties',
+    'class.return-type-will-change' => 'class.returntypewillchange',
+    'class.sensitive-parameter' => 'class.sensitiveparameter',
     'function.curl-file-create' => 'curlfile.construct',
 ];
 
-if (preg_match("!^manual/([^/]+)/([^/]+)$!", $URI, $match) && isset($manual_redirections[$match[2]])) {
+if (preg_match("!^manual/([^/]+)/([^/]+)(\.php)?$!", $URI, $match) && isset($manual_redirections[$match[2]])) {
```

Comment:
I don't think that can work, since the second group would capture the trailing `.php`. A possible solution would be to make the second group ungreedy:
```suggestion
if (preg_match("!^manual/([^/]+)/([^/]+?)(\.php)?$!", $URI, $match) && isset($manual_redirections[$match[2]])) {
```
It might also make sense to use non-capturing groups (i.e. `(?:…)` instead of `(…)`) where possible.

Reply:
Ah right. I was testing the regex on regex101.com, and didn't notice that the 3rd group was not captured. However, neither enabling the ungreedy mode evaluates to the desired result, as the 2nd group only captures a single character. So I thought it's simpler to simply remove the last `.php` suffix from the URI before using it in the regex.